### PR TITLE
dkg/sync: implement await shutdown

### PR DESCRIPTION
**Pull request by dB2510**
Implements await shutdown methods on sync.Client and sync.Server

category: feature
ticket: #684

### CODE REVIEW

File: dkg/sync/client.go [hunk 1]
```diff
@@ -65,71 +67,84 @@ func (c *Client) AwaitConnected() error {
...
+	return result{
+		rtt:       time.Since(before),
+		timestamp: resp.SyncTimestamp.String(),
+	}
+}
+
+func (c *Client) send(msg *pb.MsgSync) (*pb.MsgSyncResponse, error) {
+	wb, err := proto.Marshal(msg)
+	if err != nil {
+		return nil, errors.Wrap(err, "unmarshal msg")
```
Comment (xenowits) [inline dkg/sync/client.go:109]:
```suggestion
		return nil, errors.Wrap(err, "marshal msg")
```

File: dkg/sync/server.go [hunk 2]
```diff
@@ -69,7 +70,18 @@ func (s *Server) AwaitAllConnected() error {
...
+	for len(msgs) < len(s.peers) {
+		select {
+		case <-s.ctx.Done():
+			return s.ctx.Err()
+		case msg := <-s.receiveShutdown:
+			msgs = append(msgs, msg)
+		}
+	}
+
+	log.Info(s.ctx, "All Clients Shutdown Successfully 🎉", z.Any("clients", len(msgs)))
```
Comment (xenowits) [inline dkg/sync/server.go:84]:
```suggestion
	log.Info(s.ctx, "All clients shutdown successfully 🎉", z.Any("clients", len(msgs)))
```

File: dkg/sync/server.go [hunk 3]
```diff
@@ -124,6 +137,28 @@ func NewServer(ctx context.Context, tcpNode host.Host, peers []p2p.Peer, defHash
...
+				resBytes, err := proto.Marshal(resp)
+				if err != nil {
+					log.Error(ctx, "Marshal server response", err)
+					return
+				}
+
+				_, err = s.Write(resBytes)
+				if err != nil {
+					log.Error(ctx, "Send response to client", err, z.Any("client", p2p.PeerName(pID)))
+					return
```
Comment (xenowits) [inline dkg/sync/server.go:155]: discussion: why not return the `error` as well?

**Review by xenowits — APPROVED** [review]

File: dkg/sync/server.go (hunk 3 above)
Comment (dB2510) [reply dkg/sync/server.go:155]: stream handler function doesn't return error, that's why only log is there

**Pull request merged**